src/storage/src/storage.rs

```diff
@@ -121,8 +121,8 @@ pub enum ChecksumMismatch {
 #[non_exhaustive]
 pub enum ReadError {
     /// The calculated crc32c did not match server provided crc32c.
-    #[error("bad CRC on read: got {got}, want {want}")]
-    BadCrc { got: u32, want: u32 },
+    #[error("checksum mismatch")]
+    ChecksumMismatch(crate::ChecksumMismatch),
```

Comment:
@coryan - #2821 is removing this type. Do you think we should enumerate the mismatches here too?